x-pack/plugins/security_solution/public/flyout/document_details/shared/components/insight_distribution_bar.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { css } from '@emotion/css';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiBadge,
+  useEuiTheme,
+  useEuiFontSize,
+  type EuiFlexGroupProps,
+} from '@elastic/eui';
+import { DistributionBar } from '@kbn/security-solution-distribution-bar';
+
+const COUNT_LIMIT = 999;
+
+export interface InsightDistributionBarProps {
+  /**
+   * Title of the insight
+   */
+  title: string | React.ReactNode;
+  /**
+   * Distribution stats to display
+   */
+  stats: Array<{ key: string; count: number; color: string; label?: React.ReactNode }>;
+  /**
+   * Count to be displayed in the badge
+   */
+  count: number;
+  /**
+   * Flex direction of the component
+   */
+  direction?: EuiFlexGroupProps['direction'];
+  /**
+   * Optional test id
+   */
+  ['data-test-subj']?: string;
+}
+
+// Displays a distribution bar with a count badge
+export const InsightDistributionBar: React.FC<InsightDistributionBarProps> = ({
+  title,
+  stats,
+  count,
+  direction = 'row',
+  'data-test-subj': dataTestSubj,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const xsFontSize = useEuiFontSize('xs').fontSize;
+
+  const countStr = useMemo(() => {
+    return count >= COUNT_LIMIT ? `${COUNT_LIMIT}+` : `${count}`;
+  }, [count]);
+
+  return (
+    <EuiFlexGroup direction={direction} data-test-subj={dataTestSubj} responsive={false}>
+      <EuiFlexItem>
+        <EuiText
+          css={css`
+            font-size: ${xsFontSize};
+            font-weight: ${euiTheme.font.weight.bold};
+          `}
+        >
+          {title}
+        </EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFlexGroup gutterSize="xs">
```

Comment:
let add `responsive={false}` here so we guarantee this look
![Screenshot 2024-10-09 at 4 26 05 PM](https://github.com/user-attachments/assets/3dc213a8-28d2-4827-bbb2-bbcf13f7eaf3)
instead of this one
![Screenshot 2024-10-09 at 4 26 28 PM](https://github.com/user-attachments/assets/b1ce2bbc-f2e8-4154-9c66-3c6cc153ff07)